src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -147,17 +147,18 @@ protected override void PostPersist(ApplicationEngine engine)
 
                 //Remove the request from storage
                 StorageKey key = CreateStorageKey(Prefix_Request).Add(response.Id);
-                OracleRequest request = engine.Snapshot.Storages[key].GetInteroperable<OracleRequest>();
+                OracleRequest request = engine.Snapshot.Storages.TryGet(key)?.GetInteroperable<OracleRequest>();
+                if (request == null) continue;
                 engine.Snapshot.Storages.Delete(key);
 
                 //Remove the id from IdList
                 key = CreateStorageKey(Prefix_IdList).Add(GetUrlHash(request.Url));
                 IdList list = engine.Snapshot.Storages.GetAndChange(key).GetInteroperable<IdList>();
-                if (!list.Remove(response.Id)) throw new InvalidOperationException();
+                if (!list.Remove(response.Id)) continue;
```

Comment:
If the request exists, but it is not found in the `IdList`, this shows that a serious bug has occurred. I think we better stop at this time.

Reply:
If two oracle responses comes for the same request, I think that we can avoid this error and do nothing for the second one. It's a protection in case of memory pool fail.

Reply:
If two oracle responses comes for the same request, the first response will remove the request, and it will `continue` when process the second one.

Reply:
https://github.com/neo-project/neo/pull/2074/files#diff-54c2018de09b100cef60768fa7572f89e132dfb6d7c15d5674f11ae34744f7d2R151  will protect this case.

Reply:
Reverted